main.go

```diff
@@ -518,29 +526,35 @@ func postAssignee(c *gin.Context, db *sql.DB) {
 		return
 	}
 
-	for _, assigneeUsername := range assignee.Usernames {
-		assigneeUser, err := GetUserByUserName(db, assigneeUsername)
-		if err != nil {
-			fmt.Println(err)
-			c.AbortWithStatus(http.StatusInternalServerError)
-			return
-		}
+	if authorizedUserId == currentRepo.UserId {
+		for _, assigneeUsername := range assignee.Usernames {
+			assigneeUser, err := GetUserByUserName(db, assigneeUsername)
+			if err != nil {
+				fmt.Println(err)
+				c.AbortWithStatus(http.StatusInternalServerError)
+				return
+			}
 
-		_, err = db.Exec(`INSERT INTO assignees ( issue_id , user_id ) VALUES ( $1 , $2) `, issueId, assigneeUser.ID)
-		if err != nil {
-			fmt.Println(err)
-			c.AbortWithStatus(http.StatusInternalServerError)
-			return
-		}
+			_, err = db.Exec(`INSERT INTO assignees ( issue_id , user_id ) VALUES ( $1 , $2) `, issueId, assigneeUser.ID)
+			if err != nil {
+				fmt.Println(err)
+				c.AbortWithStatus(http.StatusInternalServerError)
+				return
+			}
 
-		err = CreateNotification(db, issueId, assigneeUser.ID, repoId)
-		if err != nil {
-			fmt.Println(err)
-			c.AbortWithStatus(http.StatusInternalServerError)
-			return
+			err = CreateNotification(db, issueId, assigneeUser.ID, repoId)
+			if err != nil {
+				fmt.Println(err)
+				c.AbortWithStatus(http.StatusInternalServerError)
+				return
+			}
 		}
+
+		c.Status(http.StatusCreated)
+
+	} else {
+		c.Status(http.StatusUnauthorized)
```

Comment:
This should be Access Denied not Unauthorized. User is authorized but does not have access to change the assignees. so 403.

Reply:
Done